### PR TITLE
Block: Tabs (with animations)

### PR DESCRIPTION
**Pull request by tmathern**
Fix https://github.com/hlxsites/caesars/issues/16

🔗 Test URLs:
- Before: N/A (new block)
- After: https://issue-16-tabs--caesars--hlxsites.hlx.page/caesars-palace/drafts/tmathern/tabs

📝 Description:
Tabs block.
As seen in https://www.caesars.com/caesars-palace/things-to-do/weddings.
Note that menus will not be a tabs block, but a different component (https://github.com/hlxsites/caesars/issues/31).

Authoring experience:
Screenshot 2023-02-27 at 3 48 02 PM

### CODE REVIEW

**Comment by tmathern** [comment]
https://github.com/hlxsites/caesars/pull/39

**Comment by tmathern** [comment]
Version without animations: https://github.com/hlxsites/caesars/pull/47

File: caesars-palace/blocks/tabs/tabs.css [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+.tabs.block .tab-image img {
+  object-fit: cover;
+}
+
+.tabs.block .tab-text {
+  display: flex;
+  flex-direction: column;
+  padding-left: 16px;
+  padding-top: 16px;
+  padding-bottom: 16px;
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.css:80]: These can now use `var(--layout-size-m)`. You could also use padding shorthand here.

eg.
```
padding: var(--layout-size-m) 0 var(--layout-size-m) var(--layout-size-m);
```

File: caesars-palace/blocks/tabs/tabs.css [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+.tabs.block .tab-text h2 {
+  text-transform: uppercase;
+  margin-bottom: 16px;
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.css:85]: --layout-size-m

File: caesars-palace/blocks/tabs/tabs.css [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+.tabs.block .tab-text h3 {
+  text-transform: uppercase;
+  margin-bottom: 16px;
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.css:90]: --layout-size-m

File: caesars-palace/blocks/tabs/tabs.css [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+.tabs.block .tab-text h4 {
+  text-transform: uppercase;
+  margin-bottom: 16px;
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.css:95]: --layout-size-m

Comment (arumsey) [reply caesars-palace/blocks/tabs/tabs.css:95]: Actually, you group all of these together.

```
.tabs.block .tab-text h2,
.tabs.block .tab-text h3,
.tabs.block .tab-text h4 {
  text-transform: uppercase;
  margin-bottom: 16px;
}
```

File: caesars-palace/blocks/tabs/tabs.css [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  .tab-title::after {
+    transition: transform 250ms cubic-bezier(0.4,0,0.2,1);
+    transform: scaleX(0);
+    width: 100%;
+    content: '';
+    display: block;
+    padding-bottom: 8px;
+    border-bottom: 3px solid;
+    margin: -3px auto 0;
+    border-color: #141414;
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.css:160]: --static-color-black

File: caesars-palace/blocks/tabs/tabs.css [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  .tab-title:hover::after {
+      transition: transform 300ms cubic-bezier(0.4,0,0.2,1);
+      transform: scaleX(1);
+  }
+
+  .tabs.block .active-tab-title {
+    padding-bottom: 4px;
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.css:169]: --layout-size-xxs

File: caesars-palace/blocks/tabs/tabs.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+const MOVE_TABS_BACK = -1;
+
+const TAB_SLIDE_IN_ANIMATION = 'active-tab-slide-in';
+const TAB_SLIDE_OUT_ANIMATION = 'active-tab-slide-out';
+
+const ACTIVE_TAB_TITLE_CLASS = 'active-tab-title';
+const HIDDEN_TAB_TITLE_CLASS = 'hidden-tab-title';
+
+const ACTIVE_TAB_CLASS = 'active-tab';
+const HIDDEN_TAB_CLASS = 'hidden-tab';
```
Comment (arumsey) [inline caesars-palace/blocks/tabs/tabs.js:8]: A pattern I like to use is grouping all my selectors or class names into a single object. If you prefer seaprate consts that is OK as well. Sometimes having the object just makes it easier to lookup values in yur IDE.

eg.
```
const classes = Object.freeze({
  tabSlideIn: 'active-tab-slide-in',
  tabSlideOut: 'active-tab-slide-out',
});
```

**Review by arumsey — APPROVED** [review]
👍

**Pull request merged**